dockstore-webservice/src/main/java/io/dockstore/webservice/languages/CWLHandler.java

```diff
@@ -105,11 +123,19 @@ public Entry parseWorkflowContent(Entry entry, String filepath, String content,
                 }
 
                 LOG.info("Repository has Dockstore.cwl");
-            } catch (YAMLException | NullPointerException ex) {
-                LOG.info("CWL file is malformed " + ex.getCause().toString());
+            } catch (YAMLException | NullPointerException | ClassCastException ex) {
+                String message;
+                if (ex.getCause() != null) {
+                    // seems to be possible to get underlying cause in some cases
+                    message = ex.getCause().toString();
+                } else {
+                    // in other cases, the above will NullPointer
+                    message = ex.toString();
+                }
+                LOG.info("CWL file is malformed " + message);
```

Comment:
Suggest LOG.error, unless this is a common/normal occurrence. Or maybe LOG.warn.

I think it's best to log the exception itself as a second param as well; sometimes the stack trace has useful info.

Reply:
I guess it's a matter of perspective. It's an error for the workflow developer, but I don't think its an error for us (i.e. people should be able to parse invalid cwl without it raising alarms on our end)